account_payment_partner/models/account_move.py

```diff
@@ -0,0 +1,140 @@
+            if self.type == "in_invoice":
+                pay_mode = self.with_context(
+                    force_company=self.company_id.id
+                ).partner_id.supplier_payment_mode_id
+                self.payment_mode_id = pay_mode
+                if (
+                    pay_mode
+                    and pay_mode.payment_type == "outbound"
+                    and pay_mode.payment_method_id.bank_account_required
+                    and self.commercial_partner_id.bank_ids
+                ):
+                    self.invoice_partner_bank_id = get_bank_id()
+                else:
+                    self.invoice_partner_bank_id = False
+
+            elif self.type == "out_invoice":
+                # No bank account assignation is done here as this is only
+                # needed for printing purposes and it can conflict with
+                # SEPA direct debit payments. Current report prints it.
+                self.payment_mode_id = self.with_context(
+                    force_company=self.company_id.id
+                ).partner_id.customer_payment_mode_id
+        else:
+            self.payment_mode_id = False
+            if self.type == "in_invoice":
+                self.invoice_partner_bank_id = False
+        return res
+
+    @api.onchange("payment_mode_id")
+    def _onchange_payment_mode_id(self):
+        pay_mode = self.payment_mode_id
+        if (
+            pay_mode
+            and pay_mode.payment_type == "outbound"
+            and not pay_mode.payment_method_id.bank_account_required
+        ):
+            self.invoice_partner_bank_id = False
+        elif not self.payment_mode_id:
+            self.invoice_partner_bank_id = False
+
+    @api.model
+    def create(self, vals):
+        """Fill the payment_mode_id from the partner if none is provided on
+        creation, using same method as upstream."""
+        onchanges = {"_onchange_partner_id": ["payment_mode_id"]}
+        for onchange_method, changed_fields in onchanges.items():
+            if any(f not in vals for f in changed_fields):
+                invoice = self.new(vals)
+                getattr(invoice, onchange_method)()
+                for field in changed_fields:
+                    if field not in vals and invoice[field]:
+                        vals[field] = invoice._fields[field].convert_to_write(
+                            invoice[field], invoice
+                        )
+        return super(AccountMove, self).create(vals)
+
+    # I think copying payment mode from invoice to refund by default
+    # is a good idea because the most common way of "paying" a refund is to
+    # deduct it on the payment of the next invoice (and OCA/bank-payment
+    # allows to have negative payment lines since March 2016)
+    @api.model
+    def _prepare_refund(
+        self, invoice, date_invoice=None, date=None, description=None, journal_id=None
+    ):
+        vals = super(AccountMove, self)._prepare_refund(
```

Comment:
Method `_prepare_refund()` does not exist anymore in V13. You may want to use method `_reverse_moves()` instead.

Reply:
Indeed, the function is renamed to _reverse_move_vals